multiqc/modules/anglerfish/anglerfish.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Anglerfish """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import json
+
+from multiqc.plots import bargraph, beeswarm, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    Anglerfish module class
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="Anglerfish",
+            anchor="Anglerfish",
+            href="https://github.com/remiolsen/anglerfish",
+            info="A tool to assess Illumina libraries sequenced on Oxford Nanopore for the purpose of quality control.",
+            doi="",
+        )
+
+        # Find and load any anglerfish reports
+        self.anglerfish_data = dict()
+
+        for f in self.find_log_files("anglerfish", filehandles=True):
+            self.parse_anglerfish_json(f)
+
+        # Filter to strip out ignored sample names
+        self.anglerfish_data = self.ignore_samples(self.anglerfish_data)
+
+        # Stop execution of the data if no anglerfish data is found.
+        if len(self.anglerfish_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.anglerfish_data)))
+
+        # Write parsed report data to a file
+        ## Parse whole JSON to save all its content
+        self.write_data_file(self.anglerfish_data, "multiqc_anglerfish")
+
+        # General Stats Table
+        self.anglerfish_general_stats_table()
+
+        # Adds section for Sample Stats Read length table/beeswarm plot
+        self.anglerfish_sample_stats()
+        # Adds section for Undetermined indexes plot
+        self.anglerfish_undetermined_index_chart()
+
+    def parse_anglerfish_json(self, f):
+        """Parse the JSON output from Anglerfish and save the summary statistics"""
+        try:
+            parsed_json = json.load(f["f"])
+        except:
+            log.warning("Could not parse Anglerfish JSON: '{}'".format(f["fn"]))
+            return
+
+        # Fetch a sample name from the command
+        s_name = f["s_name"]
+        self.add_data_source(f, s_name)
+        self.anglerfish_data[s_name] = {}
+
+        # Parse Sample Stats
+        ## Index for each sample and their reads in order to iterate without knowing sample names
+        index = 0
+        try:
+            for k in parsed_json["sample_stats"]:
+                key_list = k.keys()
+                for key in key_list:
+                    if key != "sample_name":
+                        self.anglerfish_data[s_name][key + "_{}".format(index)] = float(k[key])
```

Comment:
This is the standard way of doing format strings in Python 2 (which MultiQC was originally written in). However, since MultiQC became Python 3+ only a while back, I now prefer to write in the newer f-string syntax:

```suggestion
                        self.anglerfish_data[s_name][f"{key}_{index}"] = float(k[key])
```

I think it's usually quite a lot easier to read..

Reply:
More of these elsewhere in the file, please update throughout.

Reply:
Done